test/boost/sstable_datafile_test.cc

```diff
@@ -1719,11 +1719,12 @@ static std::vector<std::pair<sstring, dht::token>> token_generation_for_shard(un
 
     key_and_token_pair.reserve(tokens_to_generate);
     dht::default_partitioner = std::make_unique<dht::murmur3_partitioner>(smp_count, ignore_msb);
+    dht::murmur3_partitioner partitioner(smp_count, ignore_msb);
```

Comment:
How is this "partitioner" different from the default_partitioner defined right above it? Why do you need both?

Reply:
dht::default_partitioner = ... is a leak in abstraction and it's removed in the following patch. I had to fix the test first to be able to hide default_partitioner and fix the abstraction leak.